azure/datalake/store/transfer.py

```diff
@@ -405,6 +427,12 @@ def _update(self, future):
                         overwrite=self._parent._overwrite)
                     self._ffutures[merge_future] = parent
                 else:
+                    if not self._chunked and str(dst).endswith('.inprogress'):
```

Comment:
This actually needs to be separate, since there is something that needs to always happen in the else (marking the file as finished) but there is something that only needs to happen in the else in the event that there is a file with the suffix name  .inprogress. If it doesn't end with that suffix, then we just update the transfer state to 'finished'.